bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/console/LinkyCommandExtension.java

```diff
@@ -0,0 +1,118 @@
+ */
+
+@NonNullByDefault
+@Component(service = ConsoleCommandExtension.class)
+public class LinkyCommandExtension extends AbstractConsoleCommandExtension {
+
+    private static final String REPORT = "report";
+
+    private final ThingRegistry thingRegistry;
+
+    @Activate
+    public LinkyCommandExtension(final @Reference ThingRegistry thingRegistry) {
+        super("linky", "Interact with the Linky binding.");
+        this.thingRegistry = thingRegistry;
+    }
+
+    @Override
+    public void execute(String[] args, Console console) {
+        if (args.length >= 2) {
+            LinkyHandler handler = null;
+            try {
+                ThingUID thingUID = new ThingUID(args[0]);
+                Thing thing = thingRegistry.get(thingUID);
+                if ((thing != null) && (thing.getHandler() != null) && (thing.getHandler() instanceof LinkyHandler)) {
+                    handler = (LinkyHandler) thing.getHandler();
+                }
+            } catch (IllegalArgumentException e) {
+                handler = null;
+            }
+            if (handler == null) {
+                console.println("Bad thing id '" + args[0] + "'");
+                printUsage(console);
+            } else if (args[1] == REPORT) {
+                LocalDate now = LocalDate.now();
+                LocalDate start = now.minusDays(7);
+                LocalDate end = now.minusDays(1);
+                String separator = " ";
+                if (args.length >= 3) {
+                    try {
+                        start = LocalDate.parse(args[2], DateTimeFormatter.ISO_LOCAL_DATE);
+                    } catch (DateTimeParseException e) {
+                        console.println(
+                                "Invalid format for start day '" + args[2] + "'; expected format is YYYY-MM-DD");
+                        printUsage(console);
+                        return;
+                    }
+                }
+                if (args.length >= 4) {
+                    try {
+                        end = LocalDate.parse(args[3], DateTimeFormatter.ISO_LOCAL_DATE);
+                    } catch (DateTimeParseException e) {
+                        console.println("Invalid format for end day '" + args[3] + "'; expected format is YYYY-MM-DD");
+                        printUsage(console);
+                        return;
+                    }
+                }
+                if (!start.isBefore(now) || start.isAfter(end)) {
+                    console.println("Start day must be in the past and before the end day");
+                    printUsage(console);
+                    return;
+                }
+                if (end.isAfter(now.minusDays(1))) {
+                    end = now.minusDays(1);
+                }
+                if (args.length >= 5) {
+                    separator = args[4];
+                }
+                handler.reportValues(start, end, separator).forEach(console::println);
+            }
```

Comment:
Perhaps you should have an else statement here that prints usage as well?

Reply:
There is one in line 107